src/components/AuthenticationForm/LoginForm/LoginForm.js

```diff
@@ -0,0 +1,79 @@
+import axios from "axios";
+import React, { useState } from "react";
+import { Link, useLocation, useNavigate } from "react-router-dom";
+
+import { useAuth } from "../../../Context/AuthContext/AuthContext";
+
+const LoginForm = ({ setShowSignupForm }) => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [authState, dispatch] = useAuth();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const loginFormHandler = async (e) => {
+    e.preventDefault();
+
+    const response = await axios.post("api/auth/login", { email, password });
+    if (response.status === 200) {
+      dispatch({ type: "LOGIN_USER", payload: response.data });
+      localStorage.setItem("jwt", JSON.stringify(response.data.encodedToken));
+
+      navigate(location.state?.from?.pathname || "/", { replace: true });
```

Comment:
Hey, can you explain line 22? I did not understand the code here.

Reply:
I wanted to send the user to the location they were trying to visit before logging in. For example, if the user were not logged in and tried to visit the cart page, they would first be redirected to the login page using the useNaviagte hook and what the useLocation hook does is that it remebers the page that the user was trying to access before being redirected to the login page. Once the user is logged in successfully, this line of code comes into play and navigates the user to the page they were trying to visit (cart page in this example). This is only for a better UX and nothing else.
This is a great resource to learn more: https://reactrouter.com/docs/en/v6/examples/auth